src/Std/Data/Iterators/Lemmas/Producers/List.lean

```diff
@@ -0,0 +1,47 @@
+/-
+Copyright (c) 2025 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul Reichert
+-/
+prelude
+import Std.Data.Iterators.Consumers
+import Std.Data.Iterators.Lemmas.Producers.Monadic.List
+
+/-!
+# Lemmas about list iterators
+
+This module provides lemmas about the interactions of `List.iter` with `Iter.step` and various
+collectors.
+-/
+
+namespace Std.Iterators
+
+variable {β : Type w}
+
+@[simp]
+theorem _root_.List.step_iter_nil :
+    (([] : List β).iter).step = ⟨.done, rfl⟩ := by
+  simp only [Iter.step, IterM.step, Iterator.step]; rfl
+
+@[simp]
+theorem _root_.List.step_iter_cons {x : β} {xs : List β} :
+    ((x :: xs).iter).step = ⟨.yield xs.iter x, rfl⟩ := by
+  simp only [List.iter, List.iterM, IterM.step, Iterator.step]; rfl
+
+@[simp]
+theorem _root_.List.toArray_iter {m : Type w → Type w'} [Monad m] [LawfulMonad m] {l : List β} :
```

Comment:
```suggestion
theorem _root_.List.toArray_iterM {m : Type w → Type w'} [Monad m] [LawfulMonad m] {l : List β} :
```
and similar below?

Reply:
Quite the opposite -- these lemmas should have been stated for `iter`, not `iterM`. The `toArray_iterM` lemma already exists in `Std.Data.Iterators.Lemmas.Producers.Monadic.List`.

Thanks!